sdk/src/main/scala/com/horizen/account/api/http/AccountTransactionApiRoute.scala

```diff
@@ -254,14 +261,235 @@ case class AccountTransactionApiRoute(override val settings: RESTApiSettings,
+      // lock the view and try to create CoreTransaction
+      applyOnNodeView { sidechainNodeView =>
+        val valueInWei = ZenWeiConverter.convertZenniesToWei(body.forgerStakeInfo.value)
+
+        // TODO actual gas implementation
+        var maxFeePerGas = BigInteger.ONE
+        var maxPriorityFeePerGas = BigInteger.ONE
+        var gasLimit = BigInteger.ONE
+        if (body.gasInfo.isDefined) {
+          maxFeePerGas = body.gasInfo.get.maxFeePerGas
+          maxPriorityFeePerGas = body.gasInfo.get.maxPriorityFeePerGas
+          gasLimit = body.gasInfo.get.gasLimit
+        }
+        //TODO Probably getFittingSecret would need to take into account also gas
+        val secret = getFittingSecret(sidechainNodeView, None, valueInWei)
+        secret match {
+          case Some(secret) =>
+
+            val to = BytesUtils.toHexString(ForgerStakeMsgProcessor.ForgerStakeSmartContractAddress.address())
+            val nonce = body.nonce.getOrElse(sidechainNodeView.getNodeState.getNonce(secret.publicImage.address))
+            val data = encodeAddNewStakeCmdRequest(body.forgerStakeInfo)
+            val tmpTx: EthereumTransaction = new EthereumTransaction(
+              body.chainId,
+              to,
+              nonce,
+              gasLimit,
+              maxPriorityFeePerGas,
+              maxFeePerGas,
+              valueInWei,
+              data,
+              null
+            );
+            validateAndSendTransaction(signTransactionWithSecret(secret, tmpTx))
+          case None =>
+            ApiResponseUtil.toResponse(ErrorInsufficientBalance("No account with enough balance found", JOptional.empty()))
+        }
+
+      }
+    }
+  }
+
+  def spendForgingStake: Route = (post & path("spendForgingStake")) {
+    entity(as[ReqSpendForgingStake]) { body =>
+      // lock the view and try to create CoreTransaction
+      applyOnNodeView { sidechainNodeView =>
+        val valueInWei = BigInteger.ZERO
+        // TODO actual gas implementation
+        var maxFeePerGas = BigInteger.ONE
+        var maxPriorityFeePerGas = BigInteger.ONE
+        var gasLimit = BigInteger.ONE
+        if (body.gasInfo.isDefined) {
+          maxFeePerGas = body.gasInfo.get.maxFeePerGas
+          maxPriorityFeePerGas = body.gasInfo.get.maxPriorityFeePerGas
+          gasLimit = body.gasInfo.get.gasLimit
+        }
+        //TODO Probably getFittingSecret would need to take into account also gas
+        val secret = getFittingSecret(sidechainNodeView, None, valueInWei)
+        secret match {
+          case Some(secret) =>
+            val to = BytesUtils.toHexString(ForgerStakeMsgProcessor.ForgerStakeSmartContractAddress.address())
+            val nonce = body.nonce.getOrElse(sidechainNodeView.getNodeState.getNonce(secret.publicImage.address))
+            val data = encodeSpendStakeCmdRequest(body.spendForgerStakeInfo)
+            val tmpTx: EthereumTransaction = new EthereumTransaction(
+              body.chainId,
```

Comment:
it think, we should retrieve `chainId` from NetworkParams instead of from user input.
Same in other places.

Reply:
Done